synapse/storage/databases/main/end_to_end_keys.py

```diff
@@ -940,11 +942,14 @@ async def get_e2e_cross_signing_keys_bulk(
         result = await self._get_bare_e2e_cross_signing_keys_bulk(user_ids)
 
         if from_user_id:
-            result = await self.db_pool.runInteraction(
-                "get_e2e_cross_signing_signatures",
-                self._get_e2e_cross_signing_signatures_txn,
-                result,
-                from_user_id,
+            result = cast(
```

Comment:
Why do we need the cast here---because Dict is invariant?

Reply:
@squahtx Any memory of this?

Reply:
I can't remember exactly, but it's likely due to dict invariance.
`_get_e2e_cross_signing_signatures_txn` returns a `Dict[str, Optional[Dict[str, JsonDict]]]`, but this PR wants it to return a `Dict[str, Optional[***Mapping***[str, JsonDict]]]`